static/app/components/forms/controls/radioGroup.tsx

```diff
@@ -46,32 +53,51 @@ const RadioGroup = <C extends string>({
     role="radiogroup"
     aria-labelledby={label}
   >
-    {(choices || []).map(([id, name, description], index) => (
-      <RadioLineItem
-        key={index}
-        role="radio"
-        index={index}
-        aria-checked={value === id}
-        disabled={disabled}
-      >
-        <Radio
-          aria-label={id}
-          disabled={disabled}
-          checked={value === id}
-          onChange={(e: React.FormEvent<HTMLInputElement>) =>
-            !disabled && onChange(id, e)
-          }
-        />
-        <RadioLineText disabled={disabled}>{name}</RadioLineText>
-        {description && (
-          <React.Fragment>
-            {/* If there is a description then we want to have a 2x2 grid so the first column width aligns with Radio Button */}
-            <div />
-            <Description>{description}</Description>
-          </React.Fragment>
-        )}
-      </RadioLineItem>
-    ))}
+    {choices.map(([id, name, description], index) => {
+      const disabledChoice = disabledChoices.find(([choiceId]) => choiceId === id);
+      const disabled = !!disabledChoice || groupDisabled;
+      const content = (
+        <React.Fragment>
+          <RadioLineItem
+            role="radio"
+            index={index}
+            aria-checked={value === id}
+            disabled={disabled}
+          >
+            <Radio
+              aria-label={t('Select %s', name)}
+              disabled={disabled}
+              checked={value === id}
+              onChange={(e: React.FormEvent<HTMLInputElement>) =>
+                !disabled && onChange(id, e)
+              }
+            />
+            <RadioLineText disabled={disabled}>{name}</RadioLineText>
+            {description && (
+              <React.Fragment>
+                {/* If there is a description then we want to have a 2x2 grid so the first column width aligns with Radio Button */}
+                <div />
+                <Description>{description}</Description>
+              </React.Fragment>
+            )}
+          </RadioLineItem>
+        </React.Fragment>
+      );
+
+      if (!!disabledChoice?.[1]) {
+        return (
+          <Tooltip
+            key={index}
+            disabled={!disabledChoice?.[1]}
+            title={disabledChoice?.[1]}
```

Comment:
nit: Can we extract these 3 to a variable that's more descriptive? Perhaps disabledReason?